cmd/controller/main.go

```diff
@@ -93,7 +93,7 @@ func main() {
 	// Build all of our controllers, with the clients constructed above.
 	// Add new controllers to this array.
 	// You also need to modify numControllers above to match this.
-	controllers := []*kncontroller.Impl{
+	controllersArray := [...]*kncontroller.Impl{
```

Comment:
This might be clearer as `[numControllers]*kncontroller.Impl`. That will also catch the issue of `numControllers < len(controllersArray)`.

Reply:
Thanks, @grantr!
I picked the [...] form to make sure the length of the array comes strictly from the list of initializers. If we specify the length as you're suggesting, there's a risk of 0 elements at the end of the array if there are fewer initializers than the specified length inside of []. 
The [numControllers - len(controllersArray)] part of the incantation covers the case of numControllers < len(controllersArray), as that makes the difference negative.

Reply:
Makes sense, I forgot about the zero value being filled in at the end.